Example/UNUMColorPicker/ViewController.swift

```diff
@@ -15,41 +15,39 @@ class ViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
 
         //color data
         let colorData: [UIColor] = [.white, .black, .blue, .green, .yellow, .red, .cyan, .brown]
 
-        //init view
-        let bundle = Bundle(for: UNUMColorPickerView.self)
-        let nib = UINib(nibName: "UNUMColorPickerView", bundle: bundle)
-        let colorPickerView = nib.instantiate(withOwner: self, options: nil).first as! UNUMColorPickerView
+        //init viewController
+        let colorPickerViewController = UNUMColorPickerViewController(colors: colorData, initColor: nil)
```

Comment:
You'll notice the `setup` function is removed and the values provided in the initializer. This is getting closer to true dependency injection, though it still isn't as secure as I'd like it -- but this gets us to a better place nonetheless. `Setup` was really init since the viewcontroller didn't really work without it -- thus this injection here gets closer to the intentions.

You'll also notice that the nib-initialization work is moved into this initializer rather than needing to be done here. Ultimately my thought is that the calling point shouldn't have to worry about the bundle, etc. This reduced reuse (the downside) but the xib and the viewcontroller are often designed in tandem and not really to be reused separately (thus, the downside is not a big deal). This is especially the case here.